sources/org.osbuild.ostree

```diff
@@ -97,29 +92,33 @@ def download(items, cache):
         ostree("remote", "add",
                uid, url,
                *verify_args,
-               repo=repo_cache)
+               repo=self.repo_cache)
 
         for key in gpg:
             ostree("remote", "gpg-import", "--stdin", uid,
-                   repo=repo_cache, _input=key)
+                   repo=self.repo_cache, _input=key)
 
         # Transfer the commit: remote → cache
         print(f"pulling {commit}", file=sys.stderr)
-        ostree("pull", uid, commit, repo=repo_cache)
+        ostree("pull", uid, commit, repo=self.repo_cache)
 
         # Remove the temporary remotes again
         ostree("remote", "delete", uid,
-               repo=repo_cache)
-
-
-class OSTreeSource(sources.SourceService):
+               repo=self.repo_cache)
 
-    def download(self, items, cache, _options):
-        cache = os.path.join(cache, "org.osbuild.ostree")
-        os.makedirs(cache, exist_ok=True)
+    def config_download(self):
+        # Prepare the cache and the output repo
+        self.repo_cache = os.path.join(self.cache, "repo")
```

Comment:
Attribute definition outside of __init__ (pylint [W0201](https://pylint.pycqa.org/en/latest/messages/warning/attribute-defined-outside-init.html?highlight=attribute-defined-outside-init)). I think this whole code can go into `__init__`, since nothing depends on the particular download.